java-base/src/main/java/net/cryptic_game/backend/base/daemon/Function.java

```diff
@@ -47,9 +47,9 @@ public Function(final JsonObject json, final Daemon daemon) throws IllegalArgume
 
     @Override
     public JsonObject serialize() {
-        return JsonBuilder.anJSON()
-                .add("name", this.getName())
-                .add("arguments", JsonUtils.toArray(this.arguments))
+        return JsonBuilder
+                .create("name", this.getName())
+                .add("arguments", JsonUtils.toArray(this.arguments, FunctionArgument::serialize))
```

Comment:
You dont need `JsonUtils#toArray(Collection<?>)`. The new Json Builder can hadle `Collections` without any help.